pmd-apex/src/main/resources/rulesets/apex/style.xml

```diff
@@ -132,4 +132,24 @@ Many interfaces (e.g. Batch) required global modifiers in the past but don't req
     </example>
   </rule>
 
+<rule name="AvoidNonExistentAnnotations"
+          since="5.5.5"
+          message="Use of non existent annotations will lead to broken Apex code which will not compile in the future."
```

Comment:
consider adding the link to the list of valid annotations here for user reference

Reply:
On second thought, the link should be part of the description rather than the message.